api/v2/serializers/post/token_update.py

```diff
@@ -23,30 +24,61 @@ def validate(self, data):
         IF identity is found validation will:
         - Ensure that user/token produces a valid driver
         """
-        validated_data = data
-        self.validate_token_with_driver(data['provider'], data['username'], data['project_name'], data['token'])
-        return validated_data
+        logger.info("TokenUpdateSerializer:validate called")
+        # create will fail if it is an invalid token, but we need to create an identity first
+        # Although there is no apparent need for this function - it does nothing
+        # It should be here as it is part of the framework.
+        return data
 
-    def create(self, validated_data):
-        identity = self._find_identity_match(validated_data['provider'], validated_data['username'], validated_data['project_name'])
+    def create(self, data):
+        logger.info("TokenUpdateSerializer::create called")
+        identity = self._find_identity_match(data['provider'], data['username'], data.get('project_name'))
         if not identity:
-            identity = self._create_identity(validated_data['provider'], validated_data['username'], validated_data['project_name'], validated_data['token'])
-            return identity
+            identity = self._create_identity(data['provider'],
+                                             data['username'],
+                                             data.get('project_name'),
+                                             data.get('token'))
+        logger.info("cred -> ex_force_auth_token = %s" % (data.get('token')))
+        identity.update_credential(identity, 'key', data['username'], replace=True)
+        identity.update_credential(identity, 'ex_force_auth_token', data.get('token'), replace=True)
+        identity.update_credential(identity,
```

Comment:
Yes, I aliased this in the user credentials and provider credentials for the following reasons:

    1) Made it easier to debug other layers.
    2) Did not require changing the way that atmosphere currently works.
    4) user credentials will override provider credentials.